openfisca_tools/data/dataset.py

```diff
@@ -0,0 +1,205 @@
+from typing import Union
+import logging
+import os
+from pathlib import Path
+import re
+import pandas as pd
+import h5py
+import numpy as np
+
+
+class Dataset:
+    name: str = None
+    label: str = None
+    model: str = None
+    data_format: str = None
+    folder_path: str = None
+
+    # Data formats
+    TABLES = "tables"
+    ARRAYS = "arrays"
+
+    def __init__(self):
+        # Setup dataset
+        try:
+            self.folder_path = Path(self.folder_path)
+        except TypeError:
+            raise TypeError(
+                f"You tried to instantiate {self.label}, but no folder_path attribute has been provided (or an invalid path supplied)."
+            )
+
+        if not self.folder_path.exists():
+            logging.info(
+                f"Creating folder {self.folder_path} for {self.label} dataset."
+            )
+            self.folder_path.mkdir(parents=True)
+
+        assert (
+            self.name
+        ), "You tried to instantiate a Dataset object, but no name has been provided."
+        assert (
+            self.label
+        ), "You tried to instantiate a Dataset object, but no label has been provided."
+
+        if self.data_format is None:
+            if self.model is None:
+                # Assume that external (raw) datasets are a collection of tables.
```

Comment:
Would it be worth an explicit attribute denoting datasets as raw vs openfisca? Or even a subclass?